react/src/functions.tsx

```diff
@@ -374,6 +375,31 @@ export const updateTableFilter = (
     }
 };
 
+/**
+ * Get the columns currently stored by material-table.
+ */
+export const getTableColumns = (tableRef: React.MutableRefObject<any>): any[] | null => {
+    if (tableRef.current) return tableRef.current.dataManager.columns;
+    return null;
+};
```

Comment:
I think this is only used below so you can just inline it with nullish coalescing: `tableRef.current?.dataManager.columns`